src/server/conn.rs

```diff
@@ -49,9 +49,17 @@ impl JoinQueue {
             joined = self.join_next() => joined,
             _ = time::sleep(timeout) => {
                 if self.queue.len() > 1 {
-                    self.queue.swap(0, 1);
```

Comment:
Rather than moving the join handle to the next position in the queue, we move it to the back. This should always result in less contention. Especially if the connection that could not be joined within the deadline uses a websocket.